src/main/java/bio/terra/service/resourcemanagement/ResourceService.java

```diff
@@ -140,7 +142,36 @@ public GoogleBucketResource getOrCreateBucketForBigQueryScratchFile(
         projectResource,
         (GoogleRegion)
             dataset.getDatasetSummary().getStorageResourceRegion(GoogleCloudResource.BIGQUERY),
-        flightId);
+        flightId,
+        null);
+  }
+
+  /**
+   * Get or create a bucket for snapshot export files
+   *
+   * @param flightId used to lock the bucket metadata during possible creation
+   * @return a reference to the bucket as a POJO GoogleBucketResource
+   * @throws CorruptMetadataException in two cases.
+   *     <ul>
+   *       <li>if the bucket already exists, but the metadata does not AND the application property
+   *           allowReuseExistingBuckets=false.
+   *       <li>if the metadata exists, but the bucket does not
+   *     </ul>
+   */
+  public GoogleBucketResource getOrCreateBucketForSnapshotExport(Snapshot snapshot, String flightId)
+      throws InterruptedException, GoogleResourceNamingException {
+    GoogleProjectResource projectResource = snapshot.getProjectResource();
+    return bucketService.getOrCreateBucket(
```

Comment:
We may want a check for projectResource presence or check to make sure that this is in fact a GCP snapshot

Reply:
That'll happen in the flight